js/index.js

```diff
@@ -36,3 +36,77 @@ for (let i = 0; i < skills.length; i++) {
   // Append the list item element to the unordered list
   skillsList.appendChild(skill);
 }
+
+// Select the message form by name
+const messageForm = document.forms.leave_message;
+
+// Add event listener for form submit
+messageForm.addEventListener('submit', function(event) {
+  // Prevent the form from submitting normally
+  event.preventDefault();
+
+  // Retrieve the form field values
+  const name = event.target.usersName.value;
+  const email = event.target.usersEmail.value;
+  let message = event.target.usersMessage.value;
+
+  // Log the form field values to the console
+  console.log('Name:', name);
+  console.log('Email:', email);
+  console.log('Message:', message);
+
+const messageSection = document.querySelector('#messages');
+//const messageSectionHeader = document.querySelector('h2');
+const messageList = messageSection.querySelector('ul');
+
+// function hideMessageSection() {
+//   if (messageList.children.length === 0 ) {
+//     messageSection.style.display = 'none';   
+//   } else {
+//     messageSection.style.display = 'block';
+//   }
+// }
+
+// function createEditButton(){
+//   const editButton = document.createElement('button');
+//   editButton.innerText = 'edit';
+//   editButton.type = 'button';
+
+//   editButton.addEventListener('click', function(){
+//     const newMessage = prompt('enter new message: ');
+//     if (newMessage !== null) {
+//       message = newMessage;
+//     }
+//   });
+//   return editButton;
```

Comment:
These can be done without adding new nested functions.  Also, messageSection.hidden = true/false is better.  The enclosing function here is the event handler for submission of the form.  The state of hidden needs to be initialized and be updated any time the length of the list can change.

Reply:
One way to implement the edit button:

- Refactor the code with separate makeEditButton and makeRemoveButton functions outside the form callback.
- Attach a unique id to each entry (e.g. 'entry' + incremented number) when it is created
- Save the three fields for each entry in an object with the unique id as key.
- When edit is invoked, append a clone of the leave_message form to the entry and populate it from the object
- replace the innerText when the cloned form is submitted